funnel/views/commentvote.py

```diff
@@ -239,21 +319,42 @@ def delete_vote(self):
         flash(message, 'info')
         return redirect(self.obj.url_for(), code=303)
 
-    @route('report_spam', methods=['POST'])
+    @route('report_spam', methods=['GET', 'POST'])
     @requires_login
     def report_spam(self):
         csrf_form = forms.Form()
-        if csrf_form.validate_on_submit():
-            self.obj.report_spam(actor=current_auth.user)
-            flash(
-                _("The comment has been reported as spam"), 'info',
-            )
-        else:
-            flash(
-                _("There was an issue reporting this comment. Please try again"),
-                'error',
-            )
-        return redirect(self.obj.commentset.views.url(), code=303)
+        if request.method == 'POST':
+            if csrf_form.validate_on_submit():
```

Comment:
For a moment I was confused about why both conditions were being tested. Had to look a bit to notice there was POST + validation failure condition below. Suggestion: use `csrf_form.validate()` here to make it not look accidental.